src/circllhist.c

```diff
@@ -1069,6 +1111,51 @@ hist_subtract(histogram_t *tgt, const histogram_t * const *hist, int cnt) {
   return rv;
 }
 
+int
+hist_subtract_as_int64(histogram_t *tgt, const histogram_t *hist) {
+  int i, tgt_idx, src_idx;
+  int rv = 0;
+  ASSERT_GOOD_HIST(tgt);
+
+  tgt_idx = src_idx = 0;
+  if(!hist) return 0;
+  ASSERT_GOOD_HIST(hist);
+  for(i=0; i<hist->used; i++) hist_insert_raw(tgt, hist->bvs[i].bucket, 0);
+
+  while(tgt_idx < tgt->used && src_idx < hist->used) {
+    int cmp = hist_bucket_cmp(tgt->bvs[tgt_idx].bucket, hist->bvs[src_idx].bucket);
+    /* if the match, attempt to subtract, and move tgt && src fwd. */
+    if(cmp == 0) {
+      if(tgt->bvs[tgt_idx].count < hist->bvs[src_idx].count) {
+        uint64_t rev = hist->bvs[src_idx].count - tgt->bvs[tgt_idx].count;
+        if(rev > INT64_MAX) rv = -1;
+        int64_t newneg = -1 * rev;
+        memcpy(&tgt->bvs[tgt_idx].count, &newneg, sizeof(uint64_t));
+      } else {
+        tgt->bvs[tgt_idx].count = tgt->bvs[tgt_idx].count - hist->bvs[src_idx].count;
+      }
+      tgt_idx++;
+      src_idx++;
+    }
+    else if(cmp > 0) {
+      tgt_idx++;
+    }
+    else {
+      if(hist->bvs[src_idx].count > 0) rv = -1;
```

Comment:
So we don't substract values if there is no bucket in tgt? That sounds odd to me: Why not return a bucket with the negative count of src?
Also in case we had (for whatever reason) a bucket with count == 0, would be treated differently.

Reply:
We fill out tgt on new line 1123. It will have all the bins the source has.

Reply:
gotcha! That makes a lot more sense now.